src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ interface Item {
   [key: string]: any;
 }
 
-export function groupByKey(items, field) {
+export function groupByKey(items: Item[], field: string): Item {
   // write code here;
+  const result: Item = {};
+
+  items.forEach((info) => {
+    if (result[info[field]] === undefined) {
```

Comment:
We know that value is undefined or array, so we can use shorter condition
```suggestion
    if (!result[info[field]]) {
```